src/sentry/sentry_metrics/querying/api.py

```diff
@@ -553,7 +808,7 @@ def _translate_query_results(execution_results: List[QueryResult]) -> Mapping[st
     interval: Optional[int] = None
 
     # For efficiency reasons, we translate the incoming data into our custom in-memory representations.
-    intermediate_groups: Dict[GroupKey, Dict[str, GroupValue]] = {}
+    intermediate_groups: OrderedDict[GroupKey, OrderedDict[str, GroupValue]] = OrderedDict()
```

Comment:
We use an ordered dictionary, since we have to preserve the order of the values that are returned by the `QueryExecutor`.